javers-core/src/main/java/org/javers/common/reflection/ReflectionUtil.java

```diff
@@ -239,10 +239,17 @@ public static List<Type> calculateHierarchyDistance(Class<?> clazz) {
                 parents.add(current);
             }
 
-            for (Class i : current.getInterfaces()) {
-                if (!interfaces.contains(i)) {
-                    interfaces.add(i);
+            Class<?>[] currentInterfaces = current.getInterfaces();
+            while (currentInterfaces.length != 0) {
+                for (final Class<?> i : currentInterfaces) {
+                    if (!interfaces.contains(i)) {
+                        interfaces.add(i);
+                    }
                 }
+
+                currentInterfaces = Arrays.stream(currentInterfaces)
+                        .flatMap(currentInterface -> Arrays.stream(currentInterface.getInterfaces()))
+                        .toArray(Class<?>[]::new);
```

Comment:
In Java 21 `SortedMap` doesn't directly extend the `Map` interface. Instead it extends `SequencedMap` and `SequencedMap` extends `Map`. Given an interface or class, getInterfaces(), returns only the interfaces directly implemented. A the end of this method the `interfaces` variable should contains `also` the possible super interfaces.